functions/all_words.py

```
@@ -532,3 +534,156 @@ def smush_solver(center, outer_uses, spicy, first_word, words, list_len=400,
+        if all(state[i] == 0 for i in req):
+            return True
+        if state in failed:
+            return False
+        budget[0] -= 1
+        if budget[0] < 0 or time.monotonic() > deadline:
+            budget[0] = -1
+            return False
+        pivot = min((i for i in req if state[i]), key=lambda i: len(by_letter[i]))
+        options = [s for s in by_letter[pivot]
+                   if used[s] < len(groups[s])
+                   and all(c <= n for c, n in zip(s, state))]
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)